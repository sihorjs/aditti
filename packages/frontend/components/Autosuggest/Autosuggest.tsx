import React, { Component } from "react";
import classnames from "classnames";
import "./Autosuggest.less";

type Props = {
    className?: string;
};

class Autosuggest extends Component<Props> {
    render() {
        return (
            <form className={classnames("autosuggest", this.props.className)}>
                <input
                    className="autosuggest__search"
                    name="autosuggest-search"
                />
                <button
                    className="autosuggest__submit flaticon-search"
                    type="button"
                />
            </form>
        );
    }
}

export default Autosuggest;
