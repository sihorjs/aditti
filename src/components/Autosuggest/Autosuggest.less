@import url("../../less/index.less");

.autosuggest {
    display: flex;
    justify-content: center;
    align-items: center;

    &__search {
        width: 430px;
        height: 35px;
        padding: 0 10px;
        box-sizing: border-box;
        border: solid 1px @gray;
        border-right: none;
        font: 20px/35px @main-font;
        color: darken(@gray, 40%);
        border-radius: 3px 0 0 3px;
        outline: none;
    }

    &__submit {
        height: 35px;
        border: solid 1px @gray;
        border-left: none;
        color: @gray;
        border-radius: 0 3px 3px 0;
        cursor: pointer;
    }
}
