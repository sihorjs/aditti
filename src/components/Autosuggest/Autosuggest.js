import React, { Component } from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import "./Autosuggest.less";

class Autosuggest extends Component {
    render() {
        return (
            <form className={classnames("autosuggest", this.props.className)}>
                <input
                    className="autosuggest__search"
                    name="autosuggest-search"
                />
                <button
                    className="autosuggest__submit flaticon-search"
                    type="button"
                />
            </form>
        );
    }
}

Autosuggest.propTypes = {
    className: PropTypes.string
};

Autosuggest.defaultProps = {
    className: null
};

export default Autosuggest;
