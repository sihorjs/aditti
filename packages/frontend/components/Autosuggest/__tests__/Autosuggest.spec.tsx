import React from "react";
import { shallow } from "enzyme";
import Autosuggest from "components/Autosuggest";

describe("<Autosuggest />", () => {
    it("renders without crash", () => {
        const tree = shallow(<Autosuggest />);

        expect(tree).toMatchSnapshot();
    });

    it("has .autosuggest wrapper class", () => {
        const component = shallow(<Autosuggest />);

        expect(component.find(".autosuggest")).toHaveLength(1);
    });

    it("receives extra class for wrapper from props", () => {
        const className = "autosuggest-wrapper";
        const component = shallow(<Autosuggest className={className} />);

        expect(component.find(`.${className}`)).toHaveLength(1);
    });
});
