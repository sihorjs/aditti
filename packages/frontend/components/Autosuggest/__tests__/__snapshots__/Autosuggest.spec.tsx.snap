// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Autosuggest /> renders without crash 1`] = `ShallowWrapper {}`;
